primitives/src/valid_proof_of_investor/v2.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{investor_zkproof_data::v2::InvestorZKProofData, CddId, Claim, IdentityId};
+
+use confidential_identity::{
+    claim_proofs::{slice_to_scalar, Verifier},
+    cryptography_core::cdd_claim::CddId as CryptoCddId,
+    CompressedRistretto, Scalar, VerifierTrait as _,
+};
+
+// ZKProofs claims
+// =========================================================
+
+/// Data structure used to check if any of its internal claims exist in context.
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct ValidProofOfInvestor;
```

Comment:
Not sure why this type exists. The functions below could be free functions instead.
Types should not be used as modules, this ain't Java... ;)